core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -461,6 +461,10 @@ class TimeSeriesShard(val dataset: Dataset,
   def completeIndexRecovery(): Unit = {
     commitPartKeyIndexBlocking()
     startFlushingIndex() // start flushing index now that we have recovered
+    // sleep until all the recovered index entries are readable. Otherwise, duplicate partIds
+    // could be created for same partKey during ingestion. This happens when an intermittent
+    // non-ingesting partition which is not in heap re-ingests before index entry is readable.
+    Thread.sleep(storeConfig.partIndexFlushMaxDelaySeconds.seconds.toMillis + 2.seconds.toMillis)
```

Comment:
This is a bit dodgy. Is it possible to sleep in a loop, testing some condition to verify that recovery has completed?

Reply:
We could verify that the last record written is readable and loop until then. There is no other way to verify really. 
But if Lucene is adhering to its contract, it should make the records readable within the maxTime we configure it with.

We have a flush thread configured as such:

```
  private val flushThread = new ControlledRealTimeReopenThread(indexWriter,
                                                               searcherManager,
                                                               storeConfig.partIndexFlushMaxDelaySeconds,
                                                               storeConfig.partIndexFlushMinDelaySeconds)
```

Reply:
If record based verification is possible, then that's what should be done. The ControlledRealTimeReopenThread is documented as "experimental", and so I wouldn't expect it to correctly honor any contract.

Reply:
fixed